rest/src/commonMain/kotlin/builder/component/ComponentContainerBuilder.kt

```diff
@@ -0,0 +1,68 @@
+package dev.kord.rest.builder.component
+
+import dev.kord.common.annotation.KordInternal
+import dev.kord.common.entity.SeparatorSpacingSize
+import kotlin.contracts.InvocationKind
+import kotlin.contracts.contract
+
+/**
+ * A builder for an object which can contain [multiple components][ContainerComponentBuilder].
+ *
+ * @see dev.kord.rest.builder.message.MessageBuilder
+ * @see ContainerBuilder
+ */
+public interface ComponentContainerBuilder {
+    @KordInternal
+    public fun addComponent(component: ContainerComponentBuilder)
+
+    /**
+     * Adds a separator with specified [spacing].
+     */
+    public fun separator(spacing: SeparatorSpacingSize): Unit = separator {
```

Comment:
`divider: Boolean` is necessary here. Also, `divider` and `spacing` should be nullable with `null` as default value, because they can be omitted and `true` and `Small (1)` will be used respectively